tests/components/api-gateway/main_test.go

```diff
@@ -0,0 +1,209 @@
+package api_gateway
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/base64"
+	"fmt"
+	"io/fs"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/avast/retry-go"
+	"github.com/kyma-project/kyma/tests/components/api-gateway/gateway-tests/pkg/client"
+	"github.com/kyma-project/kyma/tests/components/api-gateway/gateway-tests/pkg/helpers"
+	"github.com/kyma-project/kyma/tests/components/api-gateway/gateway-tests/pkg/jwt"
+	"github.com/kyma-project/kyma/tests/components/api-gateway/gateway-tests/pkg/resource"
+	"github.com/spf13/pflag"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/clientcredentials"
+
+	"github.com/cucumber/godog"
+
+	"github.com/kyma-project/kyma/common/ingressgateway"
+
+	"github.com/vrischmann/envconfig"
+
+	"github.com/kyma-project/kyma/tests/components/api-gateway/gateway-tests/pkg/manifestprocessor"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+)
+
+func init() {
+	godog.BindCommandLineFlags("godog.", &goDogOpts)
+}
+
+func TestMain(m *testing.M) {
+	pflag.Parse()
+	goDogOpts.Paths = pflag.Args()
+
+	if err := envconfig.Init(&conf); err != nil {
```

Comment:
can we split this code to seperate function, eg. PrepareTest()
Don't know if it would possible to run it as a BeforeSuite hook in InitializeApiGatewayTests

Reply:
We agreed to do it as a follow-up.